pkg/apis/argoproj/v1alpha1/argocd_types.go

```diff
@@ -316,9 +316,21 @@ type ArgoCDServerServiceSpec struct {
 	Type corev1.ServiceType `json:"type"`
 }
 
+// ArgoCDScope defines the scope of resources that ArgoCD would be managing
+type ArgoCDScope struct {
+	// When enabled lets ArgoCD manage cluster resources.
+	ClusterConfigEnabled bool `json:"clusterconfigenabled,omitempty"`
```

Comment:
I wonder if we can remove `ClusterConfigEnabled` all together.    If `Namespacce` is not empty, this feature is enabled.   If we remove `ClusterConfigEnabled`, we should rename `Namespace` to something like `ClusterConfiguredNamespaces`.

Reply:
Makes sense, Updated with a new push